Add Texas Instruments TPL5110 - Nano Power Timer, Active-Low

yankee14 opened the pull request:
This is an ultra low power timer from TI. It draws 35 nA while running.
Timing interval set with a resistor, from 100 ms to 2 h, 1% accuracy
typical. Can be triggered manually. It is designed to be used as a
high-side P-channel MOSFET driver for gating the power to a
microcontroller. It has an optional one-shot mode for enabling the
MOSFET only once.

Datasheet: http://www.ti.com/lit/ds/symlink/tpl5110.pdf

Screenshot:
![tpl5110](https://user-images.githubusercontent.com/12487802/45582593-510a0e80-b878-11e8-8049-d42dbae8c0ae.png)

Footprint: SOT-23-6*
    1. https://github.com/KiCad/kicad-footprints/blob/master/Package_TO_SOT_SMD.pretty/SOT-23-6.kicad_mod
    2. https://github.com/KiCad/kicad-footprints/blob/master/Package_TO_SOT_SMD.pretty/SOT-23-6_Handsoldering.kicad_mod

NOTE: The manufacturer's datasheet differs from the stock KiCad footprint. By the datasheet, the pad centerline distance between the two columns of pads should be 2.7 mm (no tolerance given), but it is only 2.2 mm in the stock KiCad footprint.

*Replace this line with your commit message! Please provide a description of your pull request*

------------
Thanks for creating a pull request to contribute to the KiCad libraries! To speed up integration of your PR, please check the following items:

- [ ] Provide a URL to a datasheet for the symbol(s) you are contributing
- [ ] An example screenshot image is very helpful
- [ ] Ensure that the associated footprints match the [official footprint library](https://github.com/kicad/kicad-footprints)
- [ ] If there are matching footprint PRs, provide link(s) as appropriate
- [ ] Check the output of the Travis automated check scripts - fix any errors as required

Misca1234 commented:
Thx for your contribution

> The manufacturer's datasheet differs from the stock KiCad footprint. 

Sometimes the manufactures don't agree among themselves how a standard foot print look like, but if you think the stock foot print does not match good enough you should push a manufacture specific foot print, this is not uncommon. (in this case it is 0.5 mm difference).


Please change the following 
Set the "Pin name position offset" from 40 to 20
![bild](https://user-images.githubusercontent.com/25547797/45593487-2018ff00-b988-11e8-9cce-10df64ecca00.png)

Add the SOT23 to the description , from
Timer, Nano Power, Active-High, 35 nA, 100 ms to 7200 s, VDD 1.8 V to 5.5 V, Iout max 1 mA.
into
Timer, Nano Power, Active-High, 35 nA, 100 ms to 7200 s, VDD 1.8 V to 5.5 V, Iout max 1 mA, **SOT23-6**

Pin 5 is named  "DRVn" I guess you want to indicate that it is active low, instead of adding the 'n' at the end, you add '~' at the beginning instead like this "~DRV"

Misca1234 commented:
In this case when a component have only one foot print your should explicit point it out via the field properties

![bild](https://user-images.githubusercontent.com/25547797/45593670-53a95880-b98b-11e8-998f-b35994d62a61.png)

![bild](https://user-images.githubusercontent.com/25547797/45593672-57d57600-b98b-11e8-9fdc-63af32d8ed85.png)

![bild](https://user-images.githubusercontent.com/25547797/45593674-5c019380-b98b-11e8-88f3-a56e67d06ff8.png)

![bild](https://user-images.githubusercontent.com/25547797/45593675-615ede00-b98b-11e8-8d91-775895b7e912.png)

![bild](https://user-images.githubusercontent.com/25547797/45593676-6b80dc80-b98b-11e8-97d4-85fbb719d0f4.png)

yankee14 commented:
@Misca1234 ,

>Pin 5 is named "DRVn" I guess you want to indicate that it is active low, instead of adding the 'n' at the end, you add '~' at the beginning instead like this "~DRV"

Pin 5 is named "DRV", and it is active-high, so I think it remains like it is. The TPL5111 is the one that is active-low, so I will make that change to that part.

>[Change] the SOT23 to the description...

Thank you. Fixed.

>Set the "Pin name position offset" from 40 to 20

Fixed!

On the subject of the footprint dimensions, see [discussion](https://github.com/KiCad/kicad-symbols/pull/930#issuecomment-421822342) in #930.

myfreescalewebpage commented:
Hi @yankee14 , thanks for contributing,

A few comments I have during my review:

- [ ] Footprint filter should be `SOT*23*`
- [ ] Footprint is empty, should be `Package_TO_SOT_SMD:SOT-23-6`

Else, your symbol is okay.

Cheers,
Joel

yankee14 commented:
Symbol screenshot:

![scap](https://user-images.githubusercontent.com/12487802/51635492-84dafc00-1f1c-11e9-82fb-4ab351337f8e.png)

myfreescalewebpage commented:
Thanks, symbol is correct for me now.
However, you have a file "fp-info-cache" in the git diff which is not expected... Please fix.
Cheers,
Joel

yankee14 commented:
@myfreescalewebpage wow, I did not realize I added that, sorry! It is removed.

myfreescalewebpage commented:
No worries, that's why we made reviews :)
Merging your contribution, thanks.
